framework/include/executioners/PicardSolve.h

```diff
@@ -19,166 +20,76 @@ class PicardSolve;
-   * Get the number of Picard iterations performed
-   * Because this returns the number of Picard iterations, rather than the current
-   * iteration count (which starts at 0), increment by 1.
+   * Saves the current values of the variables, and update the old(er) vectors.
    *
-   * @return Number of Picard iterations performed
+   * @param primary Whether this routine is to save the variables for the primary transformed
+   *                quantities (as main app) or the secondary ones (as a subapp)
    */
-  unsigned int numPicardIts() const { return _picard_it + 1; }
-
-  /// Check the solver status
-  MoosePicardConvergenceReason checkConvergence() const { return _picard_status; }
-
-  /// This function checks the _xfem_repeat_step flag set by solve.
-  bool XFEMRepeatStep() const { return _xfem_repeat_step; }
-
-  /// Clear Picard status
-  void clearPicardStatus() { _picard_status = MoosePicardConvergenceReason::UNSOLVED; }
-
-  /// Whether or not this has Picard iterations
-  bool hasPicardIteration() { return _has_picard_its; }
-
-  /// Set relaxation factor for the current solve as a MultiApp
-  void setMultiAppRelaxationFactor(Real factor) { _picard_self_relaxation_factor = factor; }
-
-  /// Set relaxation variables for the current solve as a MultiApp
-  void setMultiAppRelaxationVariables(const std::vector<std::string> & vars)
-  {
-    _picard_self_relaxed_variables = vars;
-  }
+  virtual void saveVariableValues(const bool primary) override final;
 
   /**
-   * Whether sub-applications are automatically advanced no matter what happens during their solves
+   * Saves the current values of the postprocessors, and update the old(er) vectors.
+   *
+   * @param primary Whether this routine is to save the variables for the primary transformed
+   *                quantities (as main app) or the secondary ones (as a subapp)
    */
-  bool autoAdvance() const;
-
-  /// mark the current solve as failed due to external conditions
-  void failStep() { _fail_step = true; }
+  virtual void savePostprocessorValues(const bool primary) override final;
 
-protected:
   /**
-   * Perform one Picard iteration or a full solve.
+   * Use the fixed point algorithm transform instead of simply using the Picard update
    *
-   * @param begin_norm_old Residual norm after timestep_begin execution of previous Picard
-   * iteration
-   * @param begin_norm     Residual norm after timestep_begin execution
-   * @param end_norm_old   Residual norm after timestep_end execution of previous Picard iteration
-   * @param end_norm       Residual norm after timestep_end execution
-   * @param relax          Whether or not we do relaxation in this iteration
-   * @param relaxed_dofs   DoFs to be relaxed
+   * @param primary Whether this routine is used for the primary transformed
+   *                quantities (as main app) or the secondary ones (as a subapp)
+   */
+  virtual bool useFixedPointAlgorithmUpdateInsteadOfPicard(const bool primary) override final;
```

Comment:
I feel like the update algorithm should be implemented  in each individual  subclass. Then we could avoid this function